From: Martin Michlmayr <tbm@cyrius.com>
Date: Fri, 2 Feb 2007 18:14:41 +0000 (+0100)
Subject: [MTD] Don't oops when the RedBoot partition table is empty
X-Git-Tag: v2.6.21-rc1^0~66^2~20
X-Git-Url: http://www2.kernel.org/git/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=678c857f3cffb076c36ce55dfa9fb570712cec12

[MTD] Don't oops when the RedBoot partition table is empty

This fixes a regression with the RedBoot parsing code introduced by
commit 0b47d654089c5ce3f2ea26a4485db9bcead1e515

Signed-off-by: Martin Michlmayr <tbm@cyrius.com>
Signed-off-by: David Woodhouse <dwmw2@infradead.org>
---

diff --git a/drivers/mtd/redboot.c b/drivers/mtd/redboot.c
index 035cd9b..5a31f49 100644
--- a/drivers/mtd/redboot.c
+++ b/drivers/mtd/redboot.c
@@ -122,7 +122,7 @@ static int parse_redboot_partitions(struct mtd_info *master,
 				}
 			}
 			break;
-		} else {
+		} else if (buf[i].size != -1) {
 			/* re-calculate of real numslots */
 			numslots = buf[i].size / sizeof(struct fis_image_desc);
 		}

